components/Event/index.js

```diff
@@ -0,0 +1,90 @@
+import { useState } from "react";
+import EventForm from "../EventForm";
+import { useRouter } from "next/router";
+import { StyledListItem } from "./Event.styled";
+import { formatDate } from "../../resources/dateUtils";
+
+export default function Event({ event, kidData, mutate }) {
+  const router = useRouter();
+
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  async function handleEditEvent(e) {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    const eventData = Object.fromEntries(formData);
+
+    const response = await fetch(`/api/events/${event?._id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(eventData),
+    });
+    if (response.ok) {
+      mutate();
+      setIsEditMode(false);
+    }
+  }
+  async function handleDeleteEvent(event) {
+    const responseEvent = await fetch(`/api/events/${event?._id}`, {
+      method: "DELETE",
+    });
+
+    if (responseEvent.ok) {
+      const responseKid = await fetch(`/api/kids/${kidData?._id}`, {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          events: kidData?.events.filter((oneEvent) => {
+            return oneEvent._id == event._id ? false : true;
+          }),
+        }),
+      });
+      if (responseKid.ok) {
+        mutate();
+      }
+    }
+  }
```

Comment:
This would be the only little thing to note. When you're in the mood sometime later, see if you can move these two functions up. That'll keep the component dumb and small =)